src/Microsoft.Health.Fhir.Shared.Api/Features/Resources/ProfileResourcesBehaviour.cs

```diff
@@ -52,12 +54,21 @@ public async Task<DeleteResourceResponse> Handle(DeleteResourceRequest request,
         private async Task<TResponse> GenericHandle<TResponse>(string resourceType, RequestHandlerDelegate<TResponse> next, CancellationToken cancellationToken)
         {
             var resources = _profilesResolver.GetProfilesTypes();
-            if (resources.Contains(resourceType) && await _authorizationService.CheckAccess(DataActions.EditProfileDefinitions, cancellationToken) != DataActions.EditProfileDefinitions)
+            if (resources.Contains(resourceType))
             {
-                throw new UnauthorizedFhirActionException();
-            }
+                if (await _authorizationService.CheckAccess(DataActions.EditProfileDefinitions, cancellationToken) != DataActions.EditProfileDefinitions)
+                {
+                    throw new UnauthorizedFhirActionException();
+                }
 
-            return await next();
+                var result = await next();
+                _profilesResolver.Refresh();
```

Comment:
nit: just for my understanding, do we expect the profile refresh to happen quickly enough to not have any impact on the response times for this request?

Reply:
It probably would add few hundred milliseconds to request, but this is for specific resources so I don't think it's critical to optimise it right now.

---
In reply to: [611309021](https://github.com/microsoft/fhir-server/pull/1795#discussion_r611309021) [](ancestors = 611309021)